### DISCUSSION TITLE
Storage: Why is the return value of from.createSignedUrl () undefined?

### DISCUSSION THREAD

**Discussion by ru88s** [supabase/supabase#1155] [Questions] [answered]
signedUrl becomes undefined.

```
const { signedUrl, data, error } = await supabase.storage.from("images").createSignedUrl(imagePath, 315360000);
console.log(signedUrl);
console.log(data);
```

![スクリーンショット 2021-04-15 11 24 09](https://user-images.githubusercontent.com/42270511/114805063-48399a80-9ddd-11eb-9889-9f8b2a2c6bc9.png)

Don't I need a signedUrl?
Or Is it a bug?

**Comment by inian** [comment, marked as answer]
Hi @ru88s, this was an error in the docs page which I have [fixed now](https://supabase.io/docs/reference/javascript/storage-from-createsignedurl). To keep it consistent with the rest of the APIs, always the value returned by the API is always inside data. So to access the signed URL, do this

```
const { data, error } = await supabase.storage.from("images").createSignedUrl(imagePath, 315360000);
console.log(data.signedURL)
```

**Reply by ru88s** [reply]
Thank you very much!

**Reply by jopfre** [reply]
I was accidentally looking at the .jp docs as they came up first in my google and they don't have the fix there. Not sure who is responsible for maintaining them?

https://www.supabase.jp/docs/reference/javascript/storage-from-createsignedurl

**Reply by inian** [reply]
ah, thats not officially maintained. We asked the person maintaining it to add a redirect to the our docs website
